### PR TITLE
test: add test coverage for api/utils package

### PR DESCRIPTION
**Pull request by paologalligit**
# Description

This commit adds a test coverage suite for the `api/utils` package.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

**Review by libotony — APPROVED** [review]

File: api/utils/http_test.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2018 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package utils
```
Comment (libotony) [inline api/utils/http_test.go:6]: Does utils_test work?

**Review by libotony — CHANGES_REQUESTED** [review]
Dismiss my previous approval reivew.

File: api/utils/http_test.go (hunk 1 above)
Comment (paologalligit) [reply api/utils/http_test.go:6]: Yes, it does. In this case I'm only using public functions from utils package. I was reading this [page](https://go.dev/doc/tutorial/add-a-test) and here it suggests to put tests in the same package as the implementation. I've realized there are different ways to approach this, do we have any convention we're following?

Comment (libotony) [reply api/utils/http_test.go:6]: I was under the impression that the package with the suffix `_test` would not be compiled into the official binary so when I was doing tests, if I don't use the private functions I would use `xxx_test` as the package name.

**Review by libotony — APPROVED** [review]

**Pull request merged**